ApplicationLibCode/FileInterface/RifVtkImportUtil.cpp

```diff
@@ -0,0 +1,192 @@
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<cvf::Vec3f> RifVtkImportUtil::readDisplacements( const pugi::xml_node& piece )
+{
+    auto pointData = piece.child( "PointData" );
+    if ( !pointData ) return {};
+
+    auto dataArray = pointData.child( "DataArray" );
+    if ( !dataArray || std::string( dataArray.attribute( "Name" ).value() ) != "disp" ) return {};
+
+    std::vector<cvf::Vec3f> displacements;
+    std::istringstream      iss( dataArray.text().get() );
+
+    double x, y, z;
+    while ( iss >> x >> y >> z )
+    {
+        displacements.emplace_back( x, y, z );
+    }
+
+    return displacements;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<unsigned> RifVtkImportUtil::readConnectivity( const pugi::xml_node& piece )
+{
+    auto cells = piece.child( "Cells" );
+    if ( !cells ) return {};
+
+    std::vector<unsigned> connectivity;
+    auto                  connectivityArray = cells.child( "DataArray" );
+
+    while ( connectivityArray )
+    {
+        if ( std::string( connectivityArray.attribute( "Name" ).value() ) == "connectivity" )
+        {
+            std::string        connectivityText = connectivityArray.text().get();
+            std::istringstream iss( connectivityText );
+
+            unsigned index;
+            while ( iss >> index )
+            {
+                connectivity.push_back( index );
+            }
+            break;
+        }
+        connectivityArray = connectivityArray.next_sibling( "DataArray" );
+    }
+
+    return connectivity;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::map<std::string, std::vector<float>> RifVtkImportUtil::readProperties( const pugi::xml_node& piece )
+{
+    auto cellData = piece.child( "CellData" );
+    if ( !cellData ) return {};
+
+    std::map<std::string, std::vector<float>> properties;
+
+    for ( const auto& dataArray : cellData.children( "DataArray" ) )
+    {
+        const char* name = dataArray.attribute( "Name" ).value();
+        if ( name && *name )
+        {
+            std::vector<float> values;
+            std::istringstream iss( dataArray.text().get() );
+
+            std::string token;
+            while ( iss >> token )
+            {
+                if ( token == "nan" || token == "NaN" || token == "NAN" )
+                {
+                    values.push_back( std::numeric_limits<float>::quiet_NaN() );
+                }
+                else
+                {
+                    try
+                    {
+                        float value = std::stof( token );
```

Comment:
This can directly be replaced by `RiaStdStringTools::toDouble`

Reply:
Done.

Reply:
Fixed.